runtime/config/types.go

```diff
@@ -0,0 +1,37 @@
+package config
+
+import (
+	"time"
+
+	"encore.dev/types/uuid"
+)
+
+// Value represents a value in the configuration for this application
+// which can be any value represented in the configuration files.
+//
+// It is a function because the underlying value could change while
+// the application is still running due to unit tests providing
+// overrides to test different behaviours.
+type Value[T any] func() T
```

Comment:
I'm still leaning towards `Dynamic[T]` (or maybe `Dyn[T]` to save on typing as it will be quite commonly used) but don't need to block on that to merge to a feature branch obviously